code/FBX/FBXConverter.cpp

```diff
@@ -2640,24 +2871,27 @@ void FBXConverter::SetShadingPropertiesRaw(aiMaterial* out_mat, const PropertyTa
                                         else {
                                             animData = animIt->second;
                                         }
+                                        
                                         for (std::pair<std::string, const AnimationCurve*> curvesIt : node->Curves()) {
                                             if (curvesIt.first == "d|DeformPercent") {
                                                 const AnimationCurve* animationCurve = curvesIt.second;
-                                                const KeyTimeList& keys = animationCurve->GetKeys();
-                                                const KeyValueList& values = animationCurve->GetValues();
+                                                
                                                 unsigned int k = 0;
-                                                for (auto key : keys) {
+                                                for (auto key_value_pair : animationCurve->GetKeyframeData()) {
```

Comment:
Converted keyframe data to a dictionary / map.

This is so much simpler to parse and understand for the average programmer now, instead of randomly linking these variables in the code, we do it once and forget about it.